### ISSUE TITLE
Add new properties to replace ILProp1::Call; and change its use so that its definition is unambiguous

### ISSUE THREAD

**Issue by NigelYiboYu** [eclipse/omr#842]
Currently, `ILProp1::Call` is used to mark an OpCode has call-like side effects. The opCode that bears this property is not necessarily a real function call. Because of this, we have to use cumbersome queries like `isFunctionCall()` in the OMR layer and J9 project layer to check for real function calls.

`OMRILOps.hpp` query

```c++
   bool isFunctionCall() const
      {
      return isCall()                           &&
             getOpCodeValue() != TR::arraycopy  &&
             getOpCodeValue() != TR::arrayset   &&
             getOpCodeValue() != TR::bitOpMem   &&
             getOpCodeValue() != TR::bitOpMemND &&
             getOpCodeValue() != TR::arraycmp   &&
             getOpCodeValue() != TR::arraycmpWithPad;
      }
```

And this is not a complete check.

We should replace ILProp1::Call with a more meaningful property name, and use ILProp1::Call only on opCodes that are function calls.

**Comment by NigelYiboYu** [comment]
@0xdaryl @andrewcraik @fjeremic This is the problem I'm facing right now: we don't have a good query to check for function calls. For my current work, I'm thinking to add a `isFunctionCal()` query in J9ILOps.hpp to override the OMR layer query, and only check for real call opCodes.

**Comment by andrewcraik** [comment]
I don't think you mean replacement in your description of what happens to the Call property. I think the ILProp1:Call can be renamed, but it should continue to apply to all the some opcodes as it currently does with a renamed isCall query on Node - basically, the first change would a be a simple and easy to verify rename.

The second change following this would be to add an isFunctionCall property (probably to somewhere other than ILProp1 because it is full) and that change is used in the codegen for the new work you are doing.

The final step would be to look at the inliner and see what what the property is that it is looking for and whether it really should be caring about isCall or isFunctionCall and to determine which it should use or if it needs a hybrid why and what property it is seeing to test.

I think a staged approach like this will be vital in helping to make sure we don't disrupt all the optimizations and codegen transformations which rely on these flags.
